act/discovery/ameriflux.py

```diff
@@ -0,0 +1,265 @@
+        default is False
+
+    Notes
+    -----
+    This programmatic interface allows users to query and automate
+    machine-to-machine downloads of Ameriflux data. This tool uses a REST URL and
+    specific parameters mentioned above and data files matching
+    the criteria will be returned to the user and downloaded.
+
+    To login/register for an Ameriflux account:
+    https://ameriflux-data.lbl.gov/Pages/RequestAccount.aspx
+
+    Examples
+    --------
+    This code will download a zip file for BASE-BADM data product at site
+    US-A37. See the Notes for information on how to obtain a username and token.
+
+    .. code-block:: python
+
+        act.discovery.download_ameriflux_data(
+            user_id, user_email, data_product="BASE-BADM", data_policy="CCBY4.0", data_variant="FULLSET",
+            site_ids=["US-A37"], agree_policy=True, intended_use="synthesis",
+            description="I intend to use this data for research", out_dir="/home/user/ameriflux_data/",
+        )
+
+    Returns
+    -------
+    files : list
+        Returns list of files retrieved
+
+    """
+    # Check all inputs are valid
+    if not isinstance(user_id, str):
+        raise ValueError("user_id should be a string...")
+
+    if not isinstance(user_email, str) or "@" not in user_email:
+        raise ValueError("user_email not a valid email...")
+
+    # Check if site_id are valid site IDs
+    check_id = _check_site_id(site_ids)
+    if isinstance(site_ids, list):
+        if any(not valid for valid in check_id):
+            warnings.warn(
+                f"{', '.join([site_ids[i] for i, valid in enumerate(check_id) if not valid])} not valid AmeriFlux Site ID",
+                UserWarning,
+            )
+            site_ids = [site_ids[i] for i, valid in enumerate(check_id) if valid]
+    elif isinstance(site_ids, str):
+        if not (check_id or site_ids in ["AA-Flx", "AA-Net"]):
+            site_ids = None
+
+    if not site_ids:
+        raise ValueError("No valid Site ID in site_ids...")
+
+    # Obtain formal intended use category
+    def intended_use_extended(intended_use):
+        return {
+            "synthesis": "Research - Multi-site synthesis",
+            "remote_sensing": "Research - Remote sensing",
+            "model": "Research - Land model/Earth system model",
+            "other_research": "Research - Other",
+            "education": "Education (Teacher or Student)",
+            "other": "Other",
+        }.get(intended_use)
+
+    if not intended_use_extended(intended_use):
+        raise ValueError("Invalid intended_use input...")
+
+    # Check if out_dir is reachable
+    if out_dir is None:
+        out_dir = os.path.abspath(os.path.expanduser("~"))
+    if not os.path.exists(out_dir):
+        raise ValueError("out_dir not valid...")
+
+    # Prompt for data policy agreement
+    if data_policy == "CCBY4.0":
+        warnings.warn(
+            "Data use guidelines for AmeriFlux CC-BY-4.0 Data Policy:\n"
```

Comment:
May want to add newlines at the start and end of this warning.

Reply:
I believe I changed the lines as requested